### PR TITLE
Add gulp-bem to build project

### PR DESCRIPTION
**Pull request by zxqfox**
Based on #135

cc @tadatuta @anton-rudeshko @voischev @blond @yeti-or

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const config = require('bem-config')();
```
Comment (blond) [inline gulpfile.js]:
```js
const path = require('path');

const config = require('bem-config')();

const gulp = require('gulp');
const bem = require('@bem/gulp');
const concat = require('gulp-concat');
const clone = require('gulp-clone');
const debug = require('gulp-debug');

const thru = require('through2');
const merge = require('merge2');

const stylus = require('gulp-stylus');
const postcss = require('gulp-postcss');
const autoprefixer = require('autoprefixer');
const postcssUrl = require('postcss-url');
const csso = require('gulp-csso');

// js
const uglify = require('gulp-uglify');

// bemhtml
const bemxjst = require('gulp-bem-xjst');

// uglification flag
const isProd = process.env.YENV === 'production';
```

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        tech: 'css',
+        extensions: ['.css', '.styl']
+    })
+    .pipe(stylus())
+    .pipe(postcss([
+        autoprefixer({
+            browsers: ['ie >= 10', 'last 2 versions', 'opera 12.1', '> 2%']
+        }),
+        postcssUrl({
+            url: (url, decl, from) =>
```
Comment (blond) [inline gulpfile.js]: Why not `url: 'inline'`?

Comment (zxqfox) [inline gulpfile.js]: Do we need inline?

File: gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+gulp.task('css', () =>
+    bundle.src({
+        tech: 'css',
+        extensions: ['.css', '.styl']
+    })
+    .pipe(stylus())
+    .pipe(postcss([
+        autoprefixer({
+            browsers: ['ie >= 10', 'last 2 versions', 'opera 12.1', '> 2%']
```
Comment (blond) [inline gulpfile.js]: May be move this options to `project-stub/browsers-support.json` config and use in Gulp and ENB?

File: gulpfile.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    .pipe(concat(bundle.name() + '.min.js'))
+    .pipe(isProd ? uglify() : skip())
+    .pipe(gulp.dest(bundle.path()))
+    .pipe(debug())
+);
+
+gulp.task('html', () => {
+    let tmpl = bundle.src({ tech: 'bemhtml.js', extensions: ['.bemhtml.js', '.bemhtml'] })
+            .pipe(thru.obj(function(file, enc, cb) {
+                if (path.basename(file.path) === 'i-bem.bemhtml') return cb(null);
```
Comment (blond) [inline gulpfile.js]: add comment with info about `bem-core` and base templates

File: gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                cb(null, file);
+            }))
+            .pipe(concat(bundle.name()))
+            .pipe(bemxjst.bemhtml()),
+
+        html = bundle.bemjson().pipe(bemxjst.toHtml(tmpl.pipe(clone())));
+
+    return merge(
+            html,
+            tmpl.pipe(clone())
```
Comment (blond) [inline gulpfile.js]: Why write BEMHTML to fs?

File: package.json [hunk 6]
```diff
@@ -13,13 +13,30 @@
     "bower": "^1.7.9"
   },
   "devDependencies": {
+    "@bem/gulp": "0.0.6",
+    "autoprefixer": "^6.3.3",
     "bem": "^0.10.2",
+    "bem-config": "1.1.0",
```
Comment (blond) [inline package.json]: `^1.1.0`?

File: gulpfile.js (hunk 5 above)
Comment (zxqfox) [inline gulpfile.js]: For debugability and using in app

File: package.json [hunk 7]
```diff
@@ -13,13 +13,30 @@
...
+    "borschik-tech-cleancss": "^2.1.0",
```
Comment (blond) [inline package.json]: Is it really need?

File: gulpfile.js (hunk 2 above)
Comment (blond) [inline gulpfile.js]: no

**Pull request closed without merging**